tools/download-kafka.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+# Copyright 2018 Ciena Corporation
+#
+# Download and extract a Kafka distribution tarball.
+# To list the supported Kafka versions:
+#
+#     tools/download-kafka.py list
+#
+# To download a given version:
+#
+#     tools/download-kafka.py get <KAFKA_VERSION>
+
+import argparse
+import hashlib
+import os
+from shutil import copyfileobj
+import sys
+import tarfile
+from tempfile import TemporaryFile
+from urllib.request import urlopen
+
+CHUNK_SIZE = 32 * 1024 * 1024
+
+versions = {
+    '0.8.0': (
+        'kafka_2.8.0-0.8.0',
+        'https://archive.apache.org/dist/kafka/0.8.0/kafka_2.8.0-0.8.0.tar.gz',
+        hashlib.md5,  # FIXME
```

Comment:
Why #FIXME?

Reply:
It looks like you got it below&mdash;md5 is no good as a signature anymore.

Reply:
Right, but why not fetch them all, check the MD5s and generate the SHA512 and put that here instead? :-)

Reply:
Well yes, the comments are a note to myself to do exactly that. :p

Reply:
I guess I was confused because the apache site has at least SHA256 for everything but 0.8.0, and if we're really paranoid, we could check the GPG signatures...

Reply:
I was looking at [this page](http://kafka.apache.org/downloads) which only seems to have GPG and md5 except for the latest release (also the file format of the md5 files is really weird -- they are not in the usual md5sum format).